inspirehep/modules/records/tasks.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Celery tasks provided by Record module."""
+
+from __future__ import absolute_import, division, print_function
+
+from celery import current_app, shared_task
+from celery.utils.log import get_task_logger
+from elasticsearch.helpers import scan
+
+from invenio_search import current_search_client as es
+from invenio_db import db
+from invenio_records.api import Record
+
+from inspire_schemas import api
+from inspirehep.modules.search.utils import dotter
+
+logger = get_task_logger(__name__)
+
+
+@shared_task(ignore_result=True)
+def replace_old_ref_for_record(old_record_ref, new_record_ref):
+    ignore_refs_in_fields = current_app.conf.get('IGNORE_REFS_IN_FIELDS')
+    schemas_names = current_app.conf.get('INSPIRE_JSON_SCHEMAS')
+    possible_fields = get_ref_fields(ignore_refs_in_fields, schemas_names)
```

Comment:
I think the correct place is in extension initialization. You could reuse the `search` one if it fits there (https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/search/ext.py#L43) which I am not sure.

Just add an extra line:

```python
self.ref_fields = get_ref_fields(ignore_refs_in_fields, schemas_names)
```

And then access it:

```python
current_app.extensions['inspire-search'].ref_fields
```

Note that the extensions get initialized in some 'random' order so you might not have all the config available at that time so the related configs can be moved next to the `ext.py` https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/search/config.py